src/parser/shaman/elemental/modules/talents/Ascendance.js

```diff
@@ -2,23 +2,40 @@ import React from 'react';
 
 import SPELLS from 'common/SPELLS';
 import SpellIcon from 'common/SpellIcon';
-import { formatPercentage } from 'common/format';
+import { formatNumber, formatPercentage } from 'common/format';
 
 import Analyzer from 'parser/core/Analyzer';
+import EnemyInstances from 'parser/shared/modules/EnemyInstances';
 
 import StatisticBox, { STATISTIC_ORDER } from 'interface/others/StatisticBox';
 
+import Abilities from '../Abilities';
+
+const ASCENDANCE_DURATION = 15000 - 1500; //remove the gcd for Ascendence itself because we only check for FS on the first cast after
+
 class Ascendance extends Analyzer {
-  numLavaBurstsCast = 0;
-  numLightningBoltsCast = 0;
-  numOtherCasts = 0;
+
+  static dependencies = {
+    abilities: Abilities,
+    enemies: EnemyInstances,
+  };
+
+  _resolveAbilityGcdField(value) {
+    if (typeof value === 'function') {
+      return value.call(this.owner, this.selectedCombatant);
+    } else {
+      return value;
+    }
+  }
+
+  justEnteredAcendance = false;
```

Comment:
Typo here (and therefore on related lines 65, 69, 70)
```suggestion
  justEnteredAscendance = false;
```